source/common/router/config_impl.cc

```diff
@@ -1249,8 +1246,7 @@ const VirtualHostImpl* RouteMatcher::findVirtualHost(const Http::RequestHeaderMa
 
   // TODO (@rshriram) Match Origin header in WebSocket
   // request with VHost, using wildcard match
-  const std::string host =
-      Http::LowerCaseString(std::string(headers.Host()->value().getStringView())).get();
+  const std::string host = Http::LowerCaseString(std::string(headers.getHostValue())).get();
```

Comment:
AFIK it's because the values aren't lower cased coming in.
Arguably we could make sure the virt host comparison is done in a case insensitive way instead but again perf -> orthogonal :-)

Reply:
Actually will add a comment to this one - I remember thinking it was a mistake the first time I saw it :-P

Reply:
can't we just lower-case without creating the intermediate? But also there is
https://github.com/abseil/abseil-cpp/blob/cf1a02e2dc5a1bc9d095f4c996306de448ca200f/absl/strings/ascii.h#L171

could be a TODO

Reply:
oh yeah, sorry, I thought you were asking why toss the lower-case-string, not why we string, lower-case, then string again.  Yeah, I can fix that :-P